pkg/reconciler/vertex/scaling/scaling.go

```diff
@@ -227,14 +228,18 @@ func (s *Scaler) scaleOneVertex(ctx context.Context, key string, worker int) err
 	// vMetrics is a map which contains metrics of all the partitions of a vertex.
 	// We need to aggregate them to get the total rate and pending of the vertex.
 	// If any of the partition doesn't have the rate or pending information, we skip scaling.
+	partitionRates := make([]float64, 0)
```

Comment:
please also add a comment on why we are also keeping individual partition data points along with the aggregates.